Update changelog for #986 (#988)
OpenLoco/OpenLoco: CHANGELOG.md

21.05+ (???)
------------------------------------------------------------------------
- Fix: [#982] Incorrect rating calculation for cargo causing penalty for fast vehicles.

> - Technical: [#986] Force align stack missing from GCC and inconsistently applied. This was causing unexplained crashes on Linux and Mac builds due to the stack not being aligned during interop hooks with loco.exe.

Reviewer: Perhaps we could shorten this a little bit, leaving the discussion to the issue:
```suggestion
- Technical: [#986] Stack misalignment in GCC builds caused unexplained crashes on Linux and Mac builds during interop hooks with loco.exe.
```
I'm wondering whether to mention #675 instead. Thoughts?

Author: Don't mind either way
